x/gov/client/cli/tx.go

```diff
@@ -140,6 +146,71 @@ $ %s tx gov submit-proposal --title="Test Proposal" --description="My awesome pr
 	return cmd
 }
 
+// NewCmdSubmitProposalV2 implements submitting a proposal transaction command.
+func NewCmdSubmitProposalV2() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "submit-proposalV2",
+		Short: "Submit a proposal along with an initial deposit",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit a proposal along with an initial deposit.
+Proposal messages and deposit can be given directly or through a proposal JSON file.
+
+Example:
+$ %s tx gov submit-proposal2 --proposal="path/to/proposal.json" --from mykey
```

Comment:
I think we'd rather use a separate json file for messages that could get generated using `--generate-only` flag and then in the code, use `authclient.ReadTxFromFile` as in `NewCmdExecAuthorization` from `x/authz`.
`parseProposalMessages` is just using `encoding/json` package which might not be enough to properly unmarshal messages with `Any`s in it.